core/templates/pages/exploration-player-page/learner-experience/input-response-pair.component.spec.ts

```diff
@@ -292,4 +292,48 @@ describe('InputResponsePairComponent', () => {
     number = '-12.3';
     expect(component.convertAnswerToLocalFormat(number)).toEqual('-12.3');
   });
+
+  it('should show loading dots when Oppia response null and last card', () => {
```

Comment:
Hm, is there a way we can write the test description such that it has a user perspective rather than focusing on implementation details? See tips here: https://github.com/oppia/oppia/wiki/Frontend-tests#name-tests-clearly

Ditto below.